src/monattack.cpp

```diff
@@ -4647,3 +4648,20 @@ bool mattack::dodge_check( monster *z, Creature *target ) {
     }
     return false;
 }
+
+bool mattack::lay_egg( monster *z )
+{
+    monster &critter = *z;
+    const furn_t &furn = g->m.furn( critter.pos() ).obj();
+    // If the monster is in a tile with the NEST flag, lay egg
+    if( furn.has_flag( TFLAG_NEST ) ) {
```

Comment:
There is `map::has_flag_furn` when you want to check only the furniture for a flag. No need to get the `furn_t` here.